olp-cpp-sdk-authentication/tests/AuthenticationClientTest.cpp

```diff
@@ -55,6 +55,80 @@ class AuthenticationClientImplTestable : public auth::AuthenticationClientImpl {
 
 ACTION_P(Wait, time) { std::this_thread::sleep_for(time); }
 
+TEST(AuthenticationClientTest, AuthenticationWithoutNetwork) {
+  auth::AuthenticationSettings settings;
+  settings.network_request_handler = nullptr;
+
+  AuthenticationClientImplTestable auth_impl(settings);
+
+  const auth::AuthenticationCredentials credentials("", "");
+
+  {
+    SCOPED_TRACE("SignUpHereUser, Offline");
+
+    auth_impl.SignUpHereUser(
+        credentials, {},
+        [=](const auth::AuthenticationClient::SignUpResponse& response) {
+          EXPECT_FALSE(response.IsSuccessful());
+          EXPECT_EQ(response.GetError().GetMessage(),
```

Comment:
I would not compare return messages. If we change or adapt them, this will fail.
Instead, please check error code.
Same below.

Reply:
done